code/modules/events/carp_migration.dm

```diff
@@ -74,3 +75,26 @@
 	if (!hasAnnounced)
 		announce_to_ghosts(fish) //Only anounce the first fish
 		hasAnnounced = TRUE
+
+/datum/event_admin_setup/carp_migration
+	/// Admin set list of turfs for carp to travel to for each z level
+	var/list/targets_per_z = list()
+
+/datum/event_admin_setup/carp_migration/prompt_admins()
+	targets_per_z = list()
+	if (tgui_alert(usr, "Direct carp to your current location? Only applies to your current Z level.", "Carp Direction", list("Yes", "No")) == "No")
+		return
```

Comment:
Rewrite this so that it respects the user closing out of the tgui_alert window (returns null), right now if it only matches for "No" it won't respect that and carry on with directing carp to the current location even if they didn't want that.